### PR TITLE
Don't call `FindMissingBlobs` when not uploading

### PR DESCRIPTION
**Pull request by brentleyjones**
If `--disk_cache` is used, along with `--incompatible_remote_results_ignore_disk` and `--noremote_upload_local_results`, results aren't uploaded to the remote cache, even if they are uploaded to the disk cache, so there is no reason to call `FindMissingBlobs`. This change removes that call when it's not needed.

### CODE REVIEW

**Comment by brentleyjones** [comment]
@coeuvre

File: src/main/java/com/google/devtools/build/lib/remote/disk/DiskAndRemoteCacheClient.java [hunk 1]
```diff
@@ -103,18 +103,22 @@ public void close() {
...
         diskCache.findMissingDigests(context, digests);
-    return Futures.whenAllSucceed(remoteQuery, diskQuery)
-        .call(
-            () ->
-                ImmutableSet.<Digest>builder()
-                    .addAll(remoteQuery.get())
-                    .addAll(diskQuery.get())
-                    .build(),
-            MoreExecutors.directExecutor());
+    if (!options.incompatibleRemoteResultsIgnoreDisk || options.remoteUploadLocalResults) {
```
Comment (coeuvre) [inline src/main/java/com/google/devtools/build/lib/remote/disk/DiskAndRemoteCacheClient.java:108]: Can we use the helper method `shouldUploadLocalResultsToRemoteCache` here?

**Review by coeuvre — CHANGES_REQUESTED** [review]

File: src/main/java/com/google/devtools/build/lib/remote/disk/DiskAndRemoteCacheClient.java (hunk 1 above)
Comment (brentleyjones) [reply src/main/java/com/google/devtools/build/lib/remote/disk/DiskAndRemoteCacheClient.java:108]: Sure. I was copying the same check that was used in the method above.

**Review by coeuvre — APPROVED** [review]
Thanks!

**Pull request closed without merging**